hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/ScheduleCompactionActionExecutor.java

```diff
@@ -109,9 +109,6 @@ public Option<HoodieCompactionPlan> execute() {
           .getWriteTimeline().filterCompletedAndCompactionInstants().getInstantsAsStream()
           .filter(instant -> compareTimestamps(instant.requestedTime(), GREATER_THAN_OR_EQUALS, instantTime))
           .collect(Collectors.toList());
-      ValidationUtils.checkArgument(conflictingInstants.isEmpty(),
```

Comment:
why are we removing it? 
and even if required, is it possible to enable it only for tbl version 8 or above?

Reply:
This was not part of table version 8 code. This was added specifically for table version 6 in https://github.com/apache/hudi/pull/12805/files

Reply:
Do we need this for backwards compatible compaction for table version 6?

Reply:
nope. this piece of code was added back by me just for table version 6. 
https://github.com/apache/hudi/pull/12805/files 

I added 2 validations as part of that patch. And @lokeshj1703  is removing the 2nd one from it. 
@lokeshj1703 : can you help us understand why do we need to remove the validation? 

If there was a test written specifically targetted for table version 8, but when you tried to run it against table version 6 and it failed -> And to get the test to succeed, if we are removing this validation, its not something that we wanted to do. 

Lets reason about table version 6 and take a call.

Reply:
This was failing due to MDT compaction scheduling with table version 6. This was not related to table version 8. 
If we delete compaction completed instant in DT but the corresponding deltacommit still exists in MDT. The MDT writer goes and tries to create compaction instant in MDT using time - one minus the inflight instant and fails check in this PR.

DT
dc10
dc20.inf
MDT
dc10
dc20
dc19 - compaction fails since dc20 already exists.

This check wouldn't have failed with older compaction scheduling logic of MDT where compaction instant is created using suffix

Reply:
We can add this check for table version 6 only after MDT backward compatible changes are in.